scripts/irods/test/test_resource_types.py

```diff
@@ -786,8 +786,12 @@ def test_detached_mode_host_not_in_host_list__issue_4421(self):
             lib.make_file(file1, 100)
 
             # put small file
-            self.admin.assert_icommand("iput -R %s %s" % (resource_name, file1), 'STDERR', 'USER_SOCK_CONNECT_ERR')  # iput
-            self.admin.assert_icommand("ils -L %s" % file1, 'STDERR_SINGLELINE', "does not exist")  # should not be listed
+            _, err, ec = self.admin.run_icommand("iput -R %s %s" % (resource_name, file1))
+            self.assertNotEqual(ec, 0)
+
+            # Both are okay, since either case is possible when connecting to an invalid host
+            # It depends on if the host rejects the connection or ignores it (i.e. a "stealth" port)
```

Comment:
Please add periods to the ends of these sentences so no one is confused later.

_Apply the same change to the other locations._